### ISSUE TITLE
rom-sql silently builds wrong join

### ISSUE THREAD

**Issue by cutalion** [rom-rb/rom-sql#331]
Surprisingly different behavior of `join(:rel)` and `join(rel)`.
rom-sql 3.0.0

```ruby
require 'rom'
require 'rom-repository'
require 'rom-sql'
require 'pry'

config = ROM::Configuration.new(:sql, 'sqlite::memory')
conn = config.gateways[:default].connection

conn.create_table(:users) do
  primary_key :id
  column :name, String
end

conn.create_table(:tasks) do
  primary_key :id
  column :user_id, Integer
  column :title, String
  column :status, String
end

config.relation(:users) do
  schema(:users, infer: true) do
    associations do
      has_many :tasks
    end
  end
end

config.relation(:tasks) do
  schema(:tasks, infer: true) do
    associations do
      belongs_to :user
    end
  end
end

rom = ROM.container(config)
users = rom.relations[:users]
tasks = rom.relations[:tasks]

user = users.changeset(:create, name: 'Alex').commit

tasks.changeset(:create, user_id: user[:id], title: 'Task 1', status: 'active').commit
tasks.changeset(:create, user_id: user[:id], title: 'Task 2', status: 'pending').commit
tasks.changeset(:create, user_id: user[:id], title: 'Task 3', status: 'pending').commit

puts 'REL'
puts users.join(tasks) { |users:, tasks:| tasks[:user_id].is(users[:id]) & tasks[:status].is('active') }

puts 'ASSOC'
puts users.join(:tasks) { |users:, tasks:| tasks[:user_id].is(users[:id]) & tasks[:status].is('active') }
```

```
REL
{:id=>1, :name=>"Alex"}
ASSOC
{:id=>1, :name=>"Alex"}
{:id=>1, :name=>"Alex"}
{:id=>1, :name=>"Alex"}
```

![image](https://user-images.githubusercontent.com/103887/57177947-3ec5fa80-6e72-11e9-974c-904e8fe46933.png)

**Comment by solnic** [comment]
That's probably a regression, damn 😢 I'll fix it ASAP.

**Comment by solnic** [comment]
@cutalion this was fixed in `v3.0.1`. It was actually an oversight in `3.0.0` where support for block-based join conditions was added.

**Comment by flash-gordon** [comment]
For the record, this was intentional to make rom-sql backwards compatible when using Sequel's block API for joining.
